### PR TITLE
Accommodate ORB product

### LINKED ISSUE #114: renaming to accomodate ORB product
Several names of configuration parameters and variable names are specific to NRB and should be renamed to something more universal to apply to ORB as well (before potentially renaming the whole repository). Here's a first list, to be implemented after a more thorough check:
- `S1_NRB.metadata.mapping`:
  - `NRB_PATTERN` -> `ARD_PATTERN` | `RB_PATTERN` | `NRB_PATTERN` + `ORB_PATTERN`?
  - `ORB_MAP` -> `OSV_MAP`
- `config.ini`:
  - `nrb_dir` -> `ard_dir` | `rb_dir`?
  - `rtc_dir` -> `sar_dir`?
  - `mode`: `all` + `rtc` + `nrb` -> `all` + `sar` + `ard`?

### PR DESCRIPTION
**Pull request by maawoo**
- Implements a hidden option: `mode = orb` (https://github.com/SAR-ARD/S1_NRB/commit/45056e9882a4f4da0185f32e362882c5241123ca)
    - Can be chosen instead of `ard`. Currently, this option has only an effect on command-line messages, file- and directory naming as well as some metadata entries. Processing steps are not changed.
- Renaming as discussed in #114:
    - `NRB_PATTERN` -> `ARD_PATTERN` (https://github.com/SAR-ARD/S1_NRB/commit/1703bca7e8c7bda183c82b193513de76ac48e4d3)
    - `ORB_MAP` -> `OSV_MAP` (https://github.com/SAR-ARD/S1_NRB/commit/dea8327fd72b770f3dab9016506ef1e5197f8d7c)
    - `nrb_dir` -> `ard_dir` (https://github.com/SAR-ARD/S1_NRB/commit/8c7b9cad46ddcfc626085c1f66aba4dbb45558f1)
    - `rtc_dir` -> `sar_dir` (https://github.com/SAR-ARD/S1_NRB/commit/de0cd1d76c64dc25d29a2392cd4dbbc52a29ac8a)
    - `mode`: `rtc` + `nrb` -> `sar` + `ard` (https://github.com/SAR-ARD/S1_NRB/commit/132bbfa1222d8ea1a9cc1fde9f6df2524cfe6f75)
- Update references to `NRB` and `RTC` to `ARD` and `SAR` where appropriate (e.g., docstrings and variable names) (https://github.com/SAR-ARD/S1_NRB/commit/c4fb7fb320fad289d8c761b0c419b09bb924aed1)
- XML: flexible namespace generation, i.e. automatically choose `s1-nrb` or `s1-orb` namespace (https://github.com/SAR-ARD/S1_NRB/commit/a9c07d19116c869ca02d465f0bcecc58f8c0c0ee)
- Update Documentation (https://github.com/SAR-ARD/S1_NRB/commit/8c4a901a297355cb28b8e133b8a8b677d2dcfe52)
- Update `metadata.stac.make_catalog` (https://github.com/SAR-ARD/S1_NRB/commit/9664cda8f741565b2128b32a558e10abd8aeaef4)

closes #114

- Merged changes of #122

### CODE REVIEW

**Comment by maawoo** [comment]
What is your opinion on merging #122 and this one soon, @johntruckenbrodt ?

File: S1_NRB/metadata/extract.py [hunk 1]
```diff
@@ -14,13 +14,13 @@
...
 import S1_NRB
-from S1_NRB.metadata.mapping import NRB_PATTERN, LERC_ERR_THRES, RES_MAP, ORB_MAP, DEM_MAP, SLC_ACC_MAP
+from S1_NRB.metadata.mapping import ARD_PATTERN, LERC_ERR_THRES, RES_MAP, OSV_MAP, DEM_MAP, SLC_ACC_MAP
 from S1_NRB import snap
 
 gdal.UseExceptions()
 
 
-def meta_dict(config, target, src_ids, rtc_dir, proc_time, start, stop, compression):
+def meta_dict(config, target, src_ids, sar_dir, proc_time, start, stop, compression, orb=False):
```
Comment (johntruckenbrodt) [inline S1_NRB/metadata/extract.py:23]: same as in https://github.com/SAR-ARD/S1_NRB/pull/122. Let's make this a generic parameter `product_type`.

File: S1_NRB/metadata/stac.py [hunk 2]
```diff
@@ -526,22 +527,24 @@ def _asset_handle_raster_ext(stac_asset, nodata, key=None, meta=None, asset=None
         raster_ext.bands[0].spatial_resolution = int(meta['prod']['demGSD'].split()[0])
 
 
-def make_catalog(directory, recursive=True, silent=False):
+def make_catalog(directory, ard_type='NRB', recursive=True, silent=False):
```
Comment (johntruckenbrodt) [inline S1_NRB/metadata/stac.py:530]: also `product_type` to be consistent

File: docs/general/folderstructure.rst [hunk 3]
```diff
@@ -3,7 +3,7 @@ Folder Structure
 
 The following demonstrates a possible structure created to store intermediate and final files during a processor run.
 The listed files describe the output of the user configuration parameter ``measurement`` set to ``gamma``
-and all output annotation layers enabled (``annotation = dm, ei, em, id, lc, li, np, gs``).
+and all output annotation layers enabled (``annotation = dm, ei, em, id, lc, li, np, gs``), thus creating S1-NRB products.
```
Comment (johntruckenbrodt) [inline docs/general/folderstructure.rst:6]: `
and all S1-NRB output annotation layers enabled (``annotation = dm, ei, em, id, lc, li, np, ratio``)
`
- `gs` has been renamed to `ratio`
- a new ORB-only layer `ld` has been introduced

**Review by johntruckenbrodt — CHANGES_REQUESTED** [review]
Looks good overall. Again, a few minor requests..

File: docs/general/folderstructure.rst (hunk 3 above)
Comment (maawoo) [reply docs/general/folderstructure.rst]: Done https://github.com/SAR-ARD/S1_NRB/pull/121/commits/a154de7b7274cf8fb809b7c087bd125ecff2cea3

Comment (johntruckenbrodt) [reply docs/general/folderstructure.rst]: The look direction angle is not intended for NRB so your sentence is a little misleading. Enabling all layers is not creating an NRB product.

Comment (maawoo) [reply docs/general/folderstructure.rst]: You're right. How about:

>and **the following** output annotation layers enabled (``annotation = dm, ei, em, id, lc, li, np, ratio``), thus creating S1-NRB products.

And only mentioning `ld` as an option in the config file, but removing it as a default.

We can integrate ORB further into the docs once it has also been further integrated in the processor.

Comment (johntruckenbrodt) [reply docs/general/folderstructure.rst]: sounds good :+1:

**Pull request merged**